crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -113,3 +174,102 @@ impl SettlementClient for EthereumSettlementClient {
         Ok(block_number.try_into()?)
     }
 }
+
+/// To prepare the sidecar for EIP 4844 transaction
+async fn prepare_sidecar(
+    state_diff: &[Vec<u8>],
+    trusted_setup: &KzgSettings,
+) -> Result<(Vec<FixedBytes<131072>>, Vec<FixedBytes<48>>, Vec<FixedBytes<48>>)> {
+    let mut sidecar_blobs = vec![];
+    let mut sidecar_commitments = vec![];
+    let mut sidecar_proofs = vec![];
+
+    for blob_data in state_diff {
+        let mut fixed_size_blob: [u8; BYTES_PER_BLOB] = [0; BYTES_PER_BLOB];
+        fixed_size_blob.copy_from_slice(blob_data.as_slice());
+
+        let blob = Blob::new(fixed_size_blob);
+
+        let commitment = KzgCommitment::blob_to_kzg_commitment(&blob, trusted_setup)?;
+        let proof = KzgProof::compute_blob_kzg_proof(&blob, &commitment.to_bytes(), trusted_setup)?;
+
+        sidecar_blobs.push(FixedBytes::new(fixed_size_blob));
+        sidecar_commitments.push(FixedBytes::new(commitment.to_bytes().into_inner()));
+        sidecar_proofs.push(FixedBytes::new(proof.to_bytes().into_inner()));
+    }
+
+    Ok((sidecar_blobs, sidecar_commitments, sidecar_proofs))
+}
+
+/// Function to construct the transaction for updating the state in core contract.
+fn get_txn_input_bytes(program_output: Vec<[u8; 32]>, kzg_proof: [u8; 48]) -> Bytes {
+    let program_output_hex_string = vec_u8_32_to_hex_string(program_output);
+    let kzg_proof_hex_string = u8_48_to_hex_string(kzg_proof);
+    // cast keccak "updateStateKzgDA(uint256[] calldata programOutput, bytes calldata kzgProof)" | cut -b 1-10
```

Comment:
nice job on adding these comments btw, it's good to have references on how to reproduce constants in code